catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java

```diff
@@ -0,0 +1,206 @@
+            for (org.opensaml.saml.saml2.core.Attribute attribute : curStatement.getAttributes()) {
+                Collection<String> attributeNames =
+                        sysHighToMetacardAttributeMapping.get(attribute.getName());
+                if (!attributeNames.isEmpty()) {
+                    for (String attributeName : attributeNames) {
+                        HashSet<Serializable> values = attribute.getAttributeValues()
+                                .stream()
+                                .filter(curValue -> curValue instanceof XSString)
+                                .map(XSString.class::cast)
+                                .map(XSString::getValue)
+                                .collect(Collectors.toCollection(HashSet::new));
+                        if (securityMarkings.get(attributeName) != null) {
+                            values.addAll(securityMarkings.get(attributeName)
+                                    .getValues());
+                        }
+                        securityMarkings.put(attributeName,
+                                new AttributeImpl(attributeName,
+                                        (List<Serializable>) ImmutableList.<Serializable>copyOf(
+                                                values)));
+                    }
+                }
+            }
+        }
+        return securityMarkings;
+    }
+
+    /**
+     * adds default system high-water markings in the event that none of the policies were able to apply security markings to this metacard
+     *
+     * @return Map of the metacard security markings to the value of their corresponding system high
+     * attribute.
+     */
+    public Metacard addDefaults(Metacard metacard) {
+
+        Map policyMap = (Map) metacard.getAttribute(Metacard.SECURITY)
+                .getValue();
+        if (metacard.getTags().contains("registry") || (policyMap != null && !policyMap.isEmpty())) {
+            return metacard;
+        }
+
+        final Metacard extendedMetacard;
+
+        if (!SECURITY_ATTRIBUTE_DESCRIPTORS.getAttributeDescriptors()
+                .stream()
+                .anyMatch(ad -> metacard.getMetacardType()
+                        .getAttributeDescriptors()
+                        .contains(ad))) {
+            MetacardImpl metacardImpl = new MetacardImpl(metacard,
+                    new MetacardTypeImpl(metacard.getMetacardType()
+                            .getName(),
+                            metacard.getMetacardType(),
+                            SECURITY_ATTRIBUTE_DESCRIPTORS.getAttributeDescriptors()));
+            Set<String> updatedTags = new HashSet<>(metacard.getTags());
+            updatedTags.add(DEFAULTMARKINGS);
+            metacardImpl.setTags(updatedTags);
+            extendedMetacard = metacardImpl;
+        } else {
+            MetacardImpl metacardImpl = new MetacardImpl(metacard);
+            Set<String> updatedTags = new HashSet<>(metacard.getTags());
```

Comment:
These four lines are identical to 133-146; they can be pulled out of the if/else.